cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/export_as_csv.spec.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Papa from 'papaparse';
+import {
+  INDEX_WITH_TIME_1,
+  PATHS,
+  DATASOURCE_NAME,
+  INDEX_WITHOUT_TIME_1,
+  DatasetTypes,
+  QueryLanguages,
+} from '../../../../../utils/constants';
+import {
+  getRandomizedWorkspaceName,
+  setDatePickerDatesAndSearchIfRelevant,
+} from '../../../../../utils/apps/query_enhancements/shared';
+import { prepareTestSuite } from '../../../../../utils/helpers';
+import {
+  downloadCsvAndVerify,
+  generateExportAsCsvIndexPatternTestConfigurations,
+  getFirstRowForDownloadWithFields,
+  getFirstRowTimeForSourceDownload,
+  getHeadersForDownloadWithFields,
+  getHeadersForSourceDownload,
+  getQueriedCountForLanguage,
+} from '../../../../../utils/apps/query_enhancements/export_as_csv';
+
+const workspaceName = getRandomizedWorkspaceName();
+
+const runSavedSearchTests = () => {
+  describe('Export as CSV', () => {
+    beforeEach(() => {
+      // Load test data
+      cy.osd.setupTestData(
+        PATHS.SECONDARY_ENGINE,
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.mapping.json`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.mapping.json`,
+        ],
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.data.ndjson`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.data.ndjson`,
+        ]
+      );
+
+      // Add data source
+      cy.osd.addDataSource({
+        name: DATASOURCE_NAME,
+        url: PATHS.SECONDARY_ENGINE,
+      });
+
+      // Create workspace
+      cy.deleteAllWorkspaces();
+      cy.visit('/app/home');
+      cy.osd.createInitialWorkspaceWithDataSource(DATASOURCE_NAME, workspaceName);
+      cy.createWorkspaceIndexPatterns({
+        workspaceName: workspaceName,
+        indexPattern: INDEX_WITH_TIME_1,
+        timefieldName: 'timestamp',
+        dataSource: DATASOURCE_NAME,
+        isEnhancement: true,
+      });
+      cy.createWorkspaceIndexPatterns({
+        workspaceName: workspaceName,
+        indexPattern: INDEX_WITHOUT_TIME_1,
+        timefieldName: '',
+        dataSource: DATASOURCE_NAME,
+        isEnhancement: true,
+        indexPatternHasTimefield: false,
+      });
```

Comment:
not for this PR: it looks like this may be a good candidate for a utility as it seems like this may be a common need across tests, is that the case or is this pretty specific?

Reply:
yeah this whole chunk can def be abstracted because its copy and pasted often times with minimal deviations. But yeah lets take that work to a different scope